### ISSUE TITLE
SvelteKit - navigator is not defined

### ISSUE THREAD

**Issue by Niceadam** [OvidijusParsiunas/deep-chat#11]
Using the Svelte demo example results in 
`ReferenceError: navigator is not defined`

navigator seems to contain information about the browser, so import has to be done `onMount` to force it on client:
```svelte
import { onMount } from 'svelte';
onMount(async () => {
  const DeepChat = await import("deep-chat");
})
```

**Comment by OvidijusParsiunas** [comment]
Hi @Niceadam, thanks for letting me know about this issue.
May I ask which versions of SvelteKit and Deep Chat are you using?
It is worth noting that a new version of Deep Chat (`1.1.21`) was released yesterday to help backwards compatibility with ES5/ES2015, it is still using `navigator`, but may be worth trying out just-in-case. If you are using the newest one could you instead try `1.1.20`. Thanks!

**Comment by OvidijusParsiunas** [comment]
I have tested this locally and the approach you are using seems to work fine for me. Here is my `+page.svelte` code:
```
<script>
	import { onMount } from 'svelte';

	onMount(async () => {
		const DeepChat = await import("deep-chat");
	});

	const introMessage = "I am your personal assistant!";
  const initialMessages = [
    { role: "user", text: "Hey, how are you today?" },
    { role: "ai", text: "I am doing very well!" }
  ];
</script>

<deep-chat
  introMessage={introMessage}
  initialMessages={JSON.stringify(initialMessages)}
  directConnection={'{"demo": true}'}
/>
```

Could you perhaps check if any of your other files are importing `deep-chat` with this syntax: `import { DeepChat } from "deep-chat";` as it seems to be giving the error you described below.

**Comment by Niceadam** [comment]
The deep-chat version didn't change anything the error was still there. I only used the import syntax above in one file, which was the demo that was up on the Svelte example in the docs.

**Comment by OvidijusParsiunas** [comment]
The CodeSandbox example that we have for Svelte is for the browser version of Svelte - not SvelteKit as the use of `import { DeepChat } from "deep-chat";` in SvelteKit would cause the `navigator/window` error.
Just to confirm, you don't have `import { DeepChat } from "deep-chat";` anywhere in your SvelteKit codebase, but the error is still coming up?

**Comment by Niceadam** [comment]
No, the error went away when I changed `import { DeepChat } from "deep-chat";` to `onMount()` solution above, the file was the only place where I imported Deepchat.

**Comment by OvidijusParsiunas** [comment]
Ohhhhhh, I understand now, I thought you meant that the error was still coming up when you were using `onMount`.
You are very correct on bringing up the issue that the Svelte example does not work in SvelteKit and I am going to create a separate CodeSandbox example now specifically for this!

**Comment by OvidijusParsiunas** [comment]
Here is a link to the new [sandbox](https://codesandbox.io/p/sandbox/deep-chat-sveltekit-fn8h6x). I will update the website + READMEs with this shortly.
Tomorrow I will start working on a full code example for hosting a SvelteKit server locally - just like the one for [NextJs](https://github.com/OvidijusParsiunas/deep-chat/tree/main/example-servers/nextjs).

Thankyou for your great feedback as we wouldn't be able to move forward without it!
If you have any other issues/feedback/suggestions, do not hesitate to create new issues and I will be more than happy to look into them.

**Comment by OvidijusParsiunas** [comment]
The example [frameworks](https://deepchat.dev/examples/frameworks) section of the [deepchat.dev](deepchat.dev) website as well as the repo's README files have been updated with the new CodeSandbox example.
If you don't mind I will close this issue as devs now have code examples for Svelte browser and SvelteKit SSR apps.
I will nevertheless update this thread once the server examples have a new SvelteKit example project.

Once again thankyou very much for your critical feedback and do not hesitate to create new issues for anything else you encounter or suggestions for better improvements!

**Comment by OvidijusParsiunas** [comment]
I have started to work on the example project but believe my current code examples are not optimal as they require the stringification of JavaScript/Array values. I have raised the following [issue](https://github.com/OvidijusParsiunas/deep-chat/issues/13) to seek for advice from devs experienced in Svelte.

**Comment by OvidijusParsiunas** [comment]
I have managed to find a fix for the problem and property values will no longer need to be stringified! I will need to test this across all other frameworks to make sure it does not break any of their functionality and should release it under a new version soon!

Explanation of the core issue:
As it turns out - Svelte automatically lower-cases the property key names that are defined by the user and because the defined names no longer match what the component is expecting they are no longer recognised as properties.

**Comment by OvidijusParsiunas** [comment]
Values no longer need to be stringified in Deep Chat version `1.2.0`.

When using SvelteKit, developers will now need to conditionally render the component on load.
The reason for this is:
When `onMount` is used SvelteKit does not recognise the Deep Chat component during its compile time, hence it will set all properties as attributes and if their values are not stringified they will come out as "[object Object]" which Deep Chat cannot parse. Hence by signaling the component to render when it has finished importing, Svelte will recongise it along with the properties which will allow their values to be passed in.
More information about this can be found in this [blog post](https://css-tricks.com/using-custom-elements-in-svelte/#:~:text=Svelte%20uses%20a%20simple%20heuristic,pass%20it%20as%20an%20attribute.).

 The live examples have also been updated to reflect this.

**Comment by OvidijusParsiunas** [comment]
Full SvelteKit code examples have been added to the repository in the [sveltekit directory](https://github.com/OvidijusParsiunas/deep-chat/tree/main/example-servers/sveltekit).
